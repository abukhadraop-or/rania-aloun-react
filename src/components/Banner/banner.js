import styled from 'styled-components';

export const BannerDiv = styled.div`
  align-items: center;
  background-color: #5cb85c;
  box-shadow: inset 0 -0.438rem 0.563rem -0.438rem rgba(0, 0, 0, 0.4);
  display: flex;
  flex-direction: column;
  height: 7rem;
  padding: 1.5rem 0;
`;

export const BannerP = styled.p`
  color: #fff;
  font-family: 'source sans pro', sans-serif;
  font-size: 1.5rem;
  margin: 0;
  text-align: center;
  @media (max-width: 375px) {
    font-size: 1.2rem;
  }
`;
