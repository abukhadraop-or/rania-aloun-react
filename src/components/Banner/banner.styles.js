import styled from 'styled-components';
import colors from 'styles/colors';
import breakpoints from 'styles/break-points';

export const Container = styled.div`
  align-items: center;
  background-color: ${colors.boringGreen};
  box-shadow: inset 0 -0.438rem 0.563rem -0.438rem rgba(0, 0, 0, 0.4),
    inset 0 7px 9px -7px rgba(0, 0, 0, 0.4);
  display: flex;
  flex-direction: column;
  height: 10.5625rem;
  padding: 1.5rem 0;
`;

export const BannerTitle = styled.p`
  color: white;
  font-family: 'titillium web', sans-serif;
  font-size: 3.5rem;
  margin: 0;
  text-shadow: 0 0 0.1rem rgba(0, 0, 0, 0.8);

  @media (max-width: ${breakpoints.mobile}) {
    font-size: 3rem;
  }
`;

export const BannerText = styled.p`
  color: white;
  font-family: 'source sans pro', sans-serif;
  font-size: 1.5rem;
  margin: 0;
  text-align: center;

  @media (max-width: ${breakpoints.mobile}) {
    font-size: 1.2rem;
  }
`;
