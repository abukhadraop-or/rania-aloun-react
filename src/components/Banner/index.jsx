import React from "react";
import { BannerDiv, BannerP } from "./banner";
import { BannerTitle } from "components/title";

/**
 * Component for showing the Banner of the home page.
 */
function index(props) {
  return (
    <BannerDiv>
      <BannerTitle>conduit</BannerTitle>
      <BannerP>A place to share your knowledge.</BannerP>
    </BannerDiv>
  );
}

export default index;
