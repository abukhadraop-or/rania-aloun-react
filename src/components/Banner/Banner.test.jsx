import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Banner from '.';

describe('<Banner />', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Banner />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
