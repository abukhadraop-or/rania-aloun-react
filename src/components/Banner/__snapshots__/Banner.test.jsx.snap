// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> matches snapshot 1`] = `
<styled.div>
  <styled.p>
    conduit
  </styled.p>
  <styled.p>
    A place to share your knowledge.
  </styled.p>
</styled.div>
`;
